eclipse-solargraph-plugin/src/main/java/io/github/pyvesb/eclipse_solargraph/launch/debug/ReadaptDebugDelegate.java

```diff
@@ -34,33 +34,43 @@
 
 import io.github.pyvesb.eclipse_solargraph.launch.RubyLaunchShortcut;
 import io.github.pyvesb.eclipse_solargraph.preferences.PreferencePage;
+import io.github.pyvesb.eclipse_solargraph.utils.ConfigHelper;
 import io.github.pyvesb.eclipse_solargraph.utils.GemHelper;
+import io.github.pyvesb.eclipse_solargraph.utils.LogHelper;
 
 public class ReadaptDebugDelegate extends DSPLaunchDelegate {
 
 	private static final AtomicBoolean HAS_UPDATED_READAPT = new AtomicBoolean();
 
+	private static final List<String> READAPT_STDIO = List.of("stdio");
+	private static final Long READAPT_UPDATE_DELAY = 5000L;
+
 	@Override
-	public void launch(ILaunchConfiguration configuration, String mode, ILaunch launch, IProgressMonitor monitor)
-			throws CoreException {
+	public void launch(ILaunchConfiguration configuration, String mode, ILaunch launch, IProgressMonitor monitor) {
 		String readaptPath = READAPT_PATH.getValue();
 		if (readaptPath == null || !new File(readaptPath).exists()) {
 			displayNotFoundWarning();
 			return;
 		}
 
 		DSPLaunchDelegateLaunchBuilder builder = new DSPLaunchDelegateLaunchBuilder(configuration, mode, launch, monitor);
-		builder.setLaunchDebugAdapter("\"" + readaptPath + "\"", List.of("stdio"));
+		builder.setLaunchDebugAdapter(readaptPath, READAPT_STDIO);
```

Comment:
The quotes were added in eda17a30ea579459a29b1e6be46e62f56eb95358, I believe they are needed. If the path contains a space in it, the command will be misinterpreted otherwise, at least on Windows.

Reply:
I noticed https://github.com/PyvesB/eclipse-solargraph/issues/21#issuecomment-1275250454, reading through that.

Reply:
I've done a bit of experimentation on both Windows and macOS, I've seen no evidence of the quotes being necessary. Though on Windows, interesting to note that it still works with or without the extra layer of quotes.